controllers/service_controller.go

```diff
@@ -247,10 +247,17 @@ func (r *ServiceReconciler) reconcileService(ctx context.Context, mms *mmesh.MMS
 	//  - when the namespace does not have the annotation modelmesh-enabled
 	//  - when the namespace is under a Terminating state.
 	n := &corev1.Namespace{}
-	if err := r.Client.Get(ctx, types.NamespacedName{Name: namespace}, n); err != nil {
-		return nil, err, false
+	if r.ClusterScope {
+		if err := r.Client.Get(ctx, types.NamespacedName{Name: namespace}, n); err != nil {
+			return nil, err, false
+		}
+	} else {
+		n = &corev1.Namespace{
```

Comment:
why reassign `n` here? Wouldn't the Client.Get call just above have returned the namespace object with the name?

Reply:
@ckadner Thanks for your review.

I have thoroughly reviewed the logic for cluster scope and namespace scope. As a result, I have made changes to ensure that this logic is executed only when in cluster scope, as in the case of namespace scope, the namespace is always the same as the controller namespace, so there is no need to retrieve the namespace object separately.

Additionally, I identified an issue where if a namespace with the "modelmesh-enabled=true" label is suddenly deleted, the code was still attempting to retrieve the namespace, causing errors in the serving runtime controller([code](https://github.com/kserve/modelmesh-serving/blob/3f018571f0dd79bb9905dde6855642c73478c048/controllers/servingruntime_controller.go#L95)). To address this, I updated [the modelmeshEnabled function](https://github.com/kserve/modelmesh-serving/pull/435/files#diff-f5aad6c746366d88daae1f63f8dcff26e2205cb9eb6b455bee29d67f0edc6573R46) in the utils.go file to consider that if the namespace does not exist, it should be treated as if modelmesh was not enabled in that namespace.

I conducted end-to-end testing (FVT) for both namespace scope and cluster scope modes locally, and both of them passed successfully.

please review it again @ckadner @tjohnson31415

Reply:
looks good now. Thanks Jooho